dvc/fs/dvc.py

```diff
@@ -209,56 +163,52 @@ def isdvc(self, path, recursive=False, strict=True):
         return bool(info.get("outs") if recurse else info.get("isout"))
 
     def info(self, path):
+        from dvc.data.meta import Meta
+
         abspath = os.path.abspath(path)
 
+        key = self._get_key(abspath)
+
         try:
-            outs = self._find_outs(abspath, strict=False, recursive=True)
-        except OutputNotFoundError as exc:
+            outs = list(self.repo.index.tree.iteritems(key))
+        except KeyError as exc:
             raise FileNotFoundError from exc
 
         ret = {
             "type": "file",
-            "outs": outs,
             "size": 0,
             "isexec": False,
             "isdvc": False,
+            "outs": outs,
         }
 
-        if len(outs) > 1:
+        if len(outs) > 1 and outs[0][0] != key:
+            shortest = self.repo.index.tree.shortest_prefix(key)
+            if shortest:
+                assert shortest[1][1].isdir
+                if len(shortest[0]) <= len(key):
+                    ret["isdvc"] = True
```

Comment:
These odd flags and logic will be removed in a followup, when we just use fs and dvcfs separately (with subrepos), instead of passing through tons of odd flags used inappropriately in ls and diff.